### PR TITLE
freebsd: replace kvm with sysctl

### PR DESCRIPTION
**Pull request by rayrapetyan**
Removed the kvm.h include, replaced the use of kvm_open() with sysctl().
No need to specify -lkvm anymore.

Similar to https://github.com/libuv/libuv/issues/1340 for OpenBSD.

### CODE REVIEW

File: src/unix/freebsd.c [hunk 1]
```diff
@@ -205,33 +204,28 @@ int uv_get_process_title(char* buffer, size_t size) {
...
+  mib[5] = 1;
 
-  kinfo = kvm_getprocs(kd, KERN_PROC_PID, pid, &nprocs);
-  if (kinfo == NULL) goto error;
+  if (sysctl(mib, 6, &kinfo, &size, NULL, 0) < 0)
+    return -errno;
 
 #ifdef __DragonFly__
-  *rss = kinfo->kp_vm_rssize * page_size;
+  *rss = kinfo.kp_vm_rssize * page_size;
```
Comment (santigimeno) [inline src/unix/freebsd.c]: Are we sure the sysctl works for `DragonFly` ?

Comment (rayrapetyan) [inline src/unix/freebsd.c]: Is there a way to specify more than one entry in AC_CHECK_LIB or I can just copy paste these lines and replace OpenBSD with FreeBSD?
Honestly I haven't tested on DragonFly.

Comment (bnoordhuis) [inline src/unix/freebsd.c]: FXR says it exists: http://fxr.watson.org/fxr/ident?v=DFBSD&i=KERN_PROC_PID

File: src/unix/freebsd.c [hunk 2]
```diff
@@ -205,33 +204,28 @@ int uv_get_process_title(char* buffer, size_t size) {
 
 
 int uv_resident_set_memory(size_t* rss) {
-  kvm_t *kd = NULL;
-  struct kinfo_proc *kinfo = NULL;
-  pid_t pid;
-  int nprocs;
+  struct kinfo_proc kinfo;
   size_t page_size = getpagesize();
+  size_t size = sizeof(struct kinfo_proc);
```
Comment (bnoordhuis) [inline src/unix/freebsd.c]: Can you use `sizeof(kinfo)` here and below and split the declarations and assignments while you're here?

**Comment by rayrapetyan** [comment]
Updated:
- netbsd is the only OS left with a kvm dependency (configure.ac and uv.gyp)
- rearranged declarations\initialization in freebsd.c

**Comment by rayrapetyan** [comment]
@santigimeno, could you describe what exactly you want to move to a separate file and why? These specific functions are already in separate files (freebsd.c\openbsd.c etc).

**Comment by santigimeno** [comment]
My suggestion was that as the `uv_resident_set_memory()` implementation is basically the same for the BSD's, it could be moved (with some #ifdef's) to a separate file a la https://github.com/libuv/libuv/blob/v1.x/src/unix/bsd-ifaddrs.c

**Comment by rayrapetyan** [comment]
Though it may be a better approach but in practice it will complicate things, e.g. I don't have ability to check such change on different BSDs. Having OS-specific code in a separate file simplifies development and allows different contributors provide updates flexibly. 
Not sure if it was a recommendation or a request, in case you're ok with provided patch please approve submitted changes.

**Comment by santigimeno** [comment]
Okay, I'm not going to block the PR as my suggestion can be implemented in the future if someone feels strongly about it.

**Comment by santigimeno** [comment]
CI: https://ci.nodejs.org/view/libuv/job/libuv-test-commit/312/

**Comment by santigimeno** [comment]
Landed in https://github.com/libuv/libuv/commit/4987b6325b40b4bb06e8597956da0f0d0c5dbcbb. Thanks!

**Pull request closed without merging**